apps/studio/src/utils/sitemap.ts

```diff
@@ -0,0 +1,101 @@
+import type { IsomerSitemap } from "@opengovsg/isomer-components"
+import type { Resource } from "@prisma/client"
+
+type ResourceDto = Omit<
+  Resource,
+  "id" | "parentId" | "publishedVersionId" | "draftBlobId"
+> & {
+  id: string
+  parentId: string | null
+}
+
+const getSitemapTreeFromArray = (
```

Comment:
not entirely sure if this is a bug, but you appear ot have missed out collections and collection pages totally, so can i trouble you to test this behaviour against those 2 items?

Reply:
also double checking here, it feels like we're retrieving the items from db in a set order (ie, at point of retrieval, the ordering is clear, which must be true because our query is centered around `parentId`) and over here, we're trying to reconstruct the sitemap again?

would it be better to instead just retain the pointers from earlier rather than doing a flat spread `[...ancestors, ...siblings, ...children] or am i misunderstanding something about the sitemap generation?

Reply:
Hmm not exactly, we don't necessarily need them to be given in a particular order for this function to work. Basically, because we are given the array, we want to convert it into a tree object and use the parentId to identify the parent-child relationships.

With the changes in bd051a9c8f9d1a82abde4ceb4c108498fb00a10a, they are now all dumped into a single array.